### PR TITLE
feat(gui): Added updateTitle() method

### PR DESCRIPTION
**Pull request by LeeGodSRC**

### CODE REVIEW

File: io.fairyproject.modules/platform.bukkit/bukkit-gui/src/main/java/io/fairyproject/bukkit/gui/Gui.java [hunk 1]
```diff
@@ -70,6 +71,33 @@ public Gui(BukkitEventNode bukkitEventNode, Component title) {
...
+
+        if (!this.isOpening() || this.titleUpdating)
+            return;
+
+        this.titleUpdating = true;
+
+        MCPlayer mcPlayer = MCPlayer.from(player);
+        player.closeInventory();
+
+        MCSchedulers.getEntityScheduler(player).schedule(() -> {
```
Comment (AquaticLabs) [inline io.fairyproject.modules/platform.bukkit/bukkit-gui/src/main/java/io/fairyproject/bukkit/gui/Gui.java]: Running a scheduler is way overkill here, by default opening a new inventory closes the previous one.

You can get away with something like
```java
    public void updateTitle(@NotNull Player player, @NotNull Component title) {
        this.title = title;

        if (!this.isOpening() || this.titleUpdating)
            return;

        this.titleUpdating = true;

        MCPlayer mcPlayer = MCPlayer.from(player);
        try {
            this.inventory = Bukkit.createInventory(null, this.getRows() * 9, MCAdventure.asLegacyString(this.title, mcPlayer.getLocale()));
            this.renderSlots(player);
            player.openInventory(this.inventory);
        } finally {
            this.titleUpdating = false;
        }
    }
```

File: io.fairyproject.modules/platform.bukkit/bukkit-gui/src/main/java/io/fairyproject/bukkit/gui/Gui.java [hunk 2]
```diff
@@ -287,6 +315,10 @@ private void onPlayerQuit(@NotNull PlayerQuitEvent event) {
 
     private void onInventoryClose(@NotNull InventoryCloseEvent event) {
         Player player = (Player) event.getPlayer();
+        // if the title is updating, the inventory is already closed
+        if (this.titleUpdating)
```
Comment (AquaticLabs) [inline io.fairyproject.modules/platform.bukkit/bukkit-gui/src/main/java/io/fairyproject/bukkit/gui/Gui.java:311]: With the changes to the updateTitle method I suggested, this is no longer true, The inventory will not be closed if the "titleUpdating"

File: io.fairyproject.modules/platform.bukkit/bukkit-gui/src/main/java/io/fairyproject/bukkit/gui/Gui.java [hunk 3]
```diff
@@ -47,13 +47,14 @@ public class Gui {
...
     private final boolean[] usedSlots;
-    private final Component title;
+    private Component title;
 
     @Nullable
     private Inventory inventory;
     private GuiSlot[] guiSlots;
     @Getter
     private EventNode<Event> eventNode;
+    private boolean titleUpdating;
```
Comment (AquaticLabs) [inline io.fairyproject.modules/platform.bukkit/bukkit-gui/src/main/java/io/fairyproject/bukkit/gui/Gui.java:57]: Not necessarily sure if this boolean is needed.

**Review by AquaticLabs — CHANGES_REQUESTED** [review]

File: io.fairyproject.modules/platform.bukkit/bukkit-gui/src/main/java/io/fairyproject/bukkit/gui/Gui.java (hunk 3 above)
Comment (LeeGodSRC) [reply io.fairyproject.modules/platform.bukkit/bukkit-gui/src/main/java/io/fairyproject/bukkit/gui/Gui.java:57]: I believe InventoryCloseEvent is still called?

File: io.fairyproject.modules/platform.bukkit/bukkit-gui/src/main/java/io/fairyproject/bukkit/gui/Gui.java (hunk 2 above)
Comment (LeeGodSRC) [reply io.fairyproject.modules/platform.bukkit/bukkit-gui/src/main/java/io/fairyproject/bukkit/gui/Gui.java:311]: I think copilot done something wrong haha

File: io.fairyproject.modules/platform.bukkit/bukkit-gui/src/main/java/io/fairyproject/bukkit/gui/Gui.java (hunk 3 above)
Comment (AquaticLabs) [reply io.fairyproject.modules/platform.bukkit/bukkit-gui/src/main/java/io/fairyproject/bukkit/gui/Gui.java:57]: Hmm, you may be correct. Will keep an eye out on that.

**Review by AquaticLabs — APPROVED** [review]
Looks good, I still believe it’d be a good idea to keep an eye on whether or not the titleUpdating boolean really does help. I did do a test with that removed and it also worked, although as you said, it could be pointlessly calling inventory.close()

**Pull request merged**